gitlab-ci: remove AutoGenerated before building

arnout opened the pull request:
We check that the AutoGenerated is unchanged after the tlvf has been
generated. However, that check doesn't take into account removed files.
To do so, remove the AutoGenerated directories before the build.

This was triggered by #1508.

arnout committed:
gitlab-ci: remove AutoGenerated before building

We check that the AutoGenerated is unchanged after the tlvf has been
generated. However, that check doesn't take into account removed files.
To do so, remove the AutoGenerated directories before the build.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>

mariomaz commented on .gitlab-ci.yml:
```diff
@@ -119,6 +119,8 @@ build-in-docker:
     EXTRA_CMAKE_FLAGS: ""
   script:
+    # To be sure that AutoGenerated is correct, remove it first and check later that it's unchanged.
+    - rm -rf framework/tlvf/AutoGenerated common/beerocks/tlvf/AutoGenerated
```
Shall we also remove `./common/beerocks/tlvf/AutoGenerated` ?

arnout replied:
It's already there:
```
    - rm -rf framework/tlvf/AutoGenerated common/beerocks/tlvf/AutoGenerated
                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

mariomaz reviewed (approved):
I've got one single comment but approving anyway because it's easy to fix in case you decide to do so.

arnout commented:
@mergifyio rebase

VladyslavTupikin commented:
@arnout Please, take a look at the CI: 3 build_in_docker jobs failed wit same log:

```
$ git update-index -q --refresh
$ git diff-index --exit-code HEAD
:100755 100644 836b7ba81b6c85e0444886b2fb0f428f4ac353cd 0000000000000000000000000000000000000000 M	framework/tlvf/AutoGenerated/include/tlvf/ieee_1905_1/cCmduHeader.h
:100644 000000 9f1e8e6b6d495c5778359665526617567ea63a26 0000000000000000000000000000000000000000 D	framework/tlvf/AutoGenerated/include/tlvf/ieee_1905_1/tlvAlMacAddressType.h
:100644 000000 2203a7c63c38278841ac7956b79c543237074ffa 0000000000000000000000000000000000000000 D	framework/tlvf/AutoGenerated/src/tlvf/ieee_1905_1/tlvAlMacAddressType.cpp
ERROR: Job failed: exit code 1
```

arnout commented:
The CI errors are exactly what this PR is supposed to do: in https://gitlab.com/prpl-foundation/prplMesh/-/jobs/625352248 the tlvAlMacAddressType.h and tlvAlMacAddressType.cpp files were still present in the repository while they actually shouldn't be there.

And now I see that there's one additional thing detected by this PR: one of the files is executable, which it obviously should't be either. I'll update this branch to fix that.

arnout committed:
tlvf: cCmduHeader.h: remove execute permission

This generated file was accidentally committed with execute permission.

Observed by removing the AutoGenerated directory and rebuilding.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
